conda/recipes/versions.yaml

```
@@ -11,7 +11,7 @@ nccl_version:
 numba_version:
   - '>=0.57'
 numpy_version:
-  - '>=1.23,<2.0a0'
+  - '>=1.23,<3.0a0'
```

```suggestion
  #- '>=1.23,<3.0a0'
  - '>=2,<3.0a0'
```

@msarahan I just updated this PR, gonna try a build with this pin to see what we can learn from the solver errors.

---

Most builds there failed in uninformative ways.

I've just pushed another set of commits that does the following:

* updates to latest `branch-24.10`, to get the `fmt` / `spdlog` updates from #722 
* commented out `cuxfiter` and `*cugraph*` packages

Hoping we can use that to confirm that only `cuxfilter` and `cugraph` remain for RAPIDS to be installable alongside NumPy 2.x.

---

🎉 that worked!

```text
...
    numpy:                            2.0.2-py311h71ddf71_0                            conda-forge 
...
```

([build link](https://github.com/rapidsai/integration/actions/runs/11017167395/job/30594424411?pr=718#step:7:966))

I think we can now say with confidence: getting `cuxfilter` and `cugraph` supporting NumPy 2 will allow RAPIDS to support NumPy 2 😁 

(cc @jakirkham @seberg for awareness)

---

`cuxfilter` was merged yesterday: https://github.com/rapidsai/cuxfilter/pull/622

So just `cugraph` remains: https://github.com/rapidsai/cugraph/pull/4615

---

I just merged https://github.com/rapidsai/cugraph/pull/4615. Will update this again once nightly packages are built from it.

Build to watch:

https://github.com/rapidsai/cugraph/actions/runs/11107406693/job/30857924045